hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+from rest_framework.response import Response
+from rest_framework import generics
+from rest_framework import serializers
+
+from hs_core.models import ResourceFile, Coverage
+
+
+# TODO: Once we upgrade past Django Rest Framework 3.3, this won't be necessary
+class JSONSerializerField(serializers.Field):
+    """ Serializer for JSONField -- required to make field writable"""
+
+    def to_internal_value(self, data):
+        return data
+    def to_representation(self, value):
+        return value
+
+
+class FileMetaDataSerializer(serializers.Serializer):
+    title = serializers.CharField()
+    keywords = JSONSerializerField()
+    spatial_coverage = JSONSerializerField()
+    extra_metadata = JSONSerializerField()
+    temporal_coverage = JSONSerializerField()
+
+
+class FileMetaDataRetrieveUpdateDestroy(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = FileMetaDataSerializer
+    allowed_methods = ('GET', 'PUT')
+
+    def get(self, request, pk, file_id):
```

Comment:
I am wondering how the API client would know if a specific file is part of a file type before requesting file type metadata as we don't have REST endpoint to provide that kind of information to the client.

Reply:
For now using ```file_id``` is probably OK but later once we implement aggregation name identifier for file type (as part of phase-2 of composite resource) we have to use aggregation name instead of file_id.

Reply:
I made an issue for this https://github.com/hydroshare/hydroshare/issues/2635